MuxExoPlayer/src/r2_13_1/java/com/mux/stats/sdk/muxstats/MuxStatsExoPlayer.java

```diff
@@ -0,0 +1,397 @@
+import com.mux.stats.sdk.core.model.CustomerVideoData;
+import com.mux.stats.sdk.core.model.CustomerViewData;
+import java.io.IOException;
+
+public class MuxStatsExoPlayer extends MuxBaseExoPlayer implements AnalyticsListener,
+    Player.EventListener {
+
+  static final String TAG = "MuxStatsEventQueue";
+
+  public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+      CustomerPlayerData customerPlayerData,
+      CustomerVideoData customerVideoData) {
+    this(ctx, player, playerName, customerPlayerData,
+        customerVideoData, null, true);
+  }
+
+  public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+      CustomerPlayerData customerPlayerData,
+      CustomerVideoData customerVideoData,
+      CustomerViewData customerViewData) {
+    this(ctx, player, playerName, customerPlayerData, customerVideoData,
+        customerViewData, true);
+  }
+
+  public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+      CustomerPlayerData customerPlayerData,
+      CustomerVideoData customerVideoData,
+      boolean sentryEnabled) {
+    this(ctx, player, playerName, customerPlayerData, customerVideoData,
+        null, sentryEnabled);
+  }
+
+  public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+      CustomerPlayerData customerPlayerData,
+      CustomerVideoData customerVideoData,
+      CustomerViewData customerViewData, boolean sentryEnabled) {
+    this(ctx, player, playerName, customerPlayerData, customerVideoData,
+        customerViewData, sentryEnabled, new MuxNetworkRequests());
+  }
+
+  public MuxStatsExoPlayer(Context ctx, ExoPlayer player, String playerName,
+      CustomerPlayerData customerPlayerData,
+      CustomerVideoData customerVideoData,
+      CustomerViewData customerViewData, boolean sentryEnabled,
+      INetworkRequest networkRequests) {
+    super(ctx, player, playerName, customerPlayerData, customerVideoData,
+        customerViewData, sentryEnabled, networkRequests);
+
+    if (player instanceof SimpleExoPlayer) {
+      ((SimpleExoPlayer) player).addAnalyticsListener(this);
+    } else {
+      player.addListener(this);
+    }
+    if (player.getPlaybackState() == Player.STATE_BUFFERING) {
+      // playback started before muxStats was initialized
+      play();
+      buffering();
+    } else if (player.getPlaybackState() == Player.STATE_READY) {
+      // We have to simulate all the events we expect to see here, even though not ideal
+      play();
+      buffering();
+      playing();
+    }
+  }
+
+  @Override
+  public void release() {
+    if (this.player != null && this.player.get() != null) {
+      ExoPlayer player = this.player.get();
+      if (player instanceof SimpleExoPlayer) {
+        ((SimpleExoPlayer) player).removeAnalyticsListener(this);
+      } else {
+        player.removeListener(this);
+      }
+    }
+    super.release();
+  }
+
+  // ------BEGIN AnalyticsListener callbacks------
+  @Override
```

Comment:
This entire file is Exact same copy of it's r2_12_1 version with exception that 
@Override
  public void onAudioSessionId(AnalyticsListener.EventTime eventTime, int audioSessionId) {
  }
  method was removed in r2.13.1 and our SDK was not using it anyway.